src/utils/JobRestAPIUtils.py

```diff
@@ -1300,9 +1321,53 @@ def UpdateEndpoints(userName, jobId, requested_endpoints, interactive_ports):
             else:
                 logger.debug("Endpoint %s exists. Skip.", endpoint_id)
 
-        dataHandler.UpdateJobTextFields({"jobId": jobId},
-                                        {"endpoints": json.dumps(endpoints)})
-        return endpoints, 200
+        is_interactive_after = is_interactive(endpoints)
+        allowed = False
+        reason = None
+
+        # is_interactive_before | is_interactive_after |  allowed
+        #         F             |        F             |     T
+        #         F             |        T             |   check
+        #         T             |        F             | impossible
+        #         T             |        T             |     T
+
+        gpu_request = get_job_total_gpu(job_params)
+
+        if is_vc_admin or \
+                gpu_request == 0 or \
+                (not (is_interactive_before is False and is_interactive_after is True)):
+            allowed = True
+        else:
+            vc_meta = json.loads(
+                walk_json(dataHandler.GetVC(job["vcName"]),
+                          "metadata",
+                          default="{}"))
+            interactive_limit = walk_json(vc_meta, "admin", "interactive_limit")
+
+            if interactive_limit is None:
+                allowed = True
+            else:
+                interactive_limit = int(interactive_limit)
+                sku = job_params["sku"]
+
+                status, _ = dataHandler.GetClusterStatus()
```

Comment:
Not sure if this will become too heavy. We can optimize later.

Reply:
Yes. I think this need to be optimized.